src/index.ts

```diff
@@ -57,18 +61,34 @@ function applyNonce(req: Request, cspString: string): string {
 
 function applyAutoTld(req: Request, cspString: string, domainOptions?: ParseOptions): string {
     if (cspString.includes(TLD)) {
-        let domain = parseDomain(req.hostname, domainOptions);
+        let tld = parseDomain(req.hostname, domainOptions);
 
-        if (!domain || !domain.tld) {
+        if (!tld) {
             return cspString;
         }
 
-        return cspString.replace(new RegExp(TLD, 'g'), domain.tld);
+        return cspString.replace(new RegExp(TLD, 'g'), tld);
     }
 
     return cspString;
 }
 
+function parseDomain(hostname: string, domainOptions?: ParseOptions): string | null {
+	for (const tld of (domainOptions?.customTlds || [])) {
```

Comment:
```suggestion
	let customTlds = domainOptions?.customTlds || [];
	for (let tld of customTlds) {
```

Reply:
This has been lifted into the `else` for the regexp support.